addons/web/models/models.py

```diff
@@ -474,7 +475,10 @@ def search_panel_select_range(self, field_name, **kwargs):
         SEARCH_PANEL_ERROR = {'error_msg': _(SEARCH_PANEL_ERROR_MESSAGE), }
 
         model_domain = kwargs.get('search_domain', [])
-        extra_domain = kwargs.get('category_domain', [])
+        extra_domain = AND([
+            kwargs.get('category_domain', []),
+            kwargs.get('filter_domain', []),
+        ])
```

Comment:
This was done like that since the beginning for the documents override that inspects the category_domain. As soon as
documents won't do that, we could indeed bring some changes to the kwargs passed. But in the end it brings an extra flexibility for other potential overrides.